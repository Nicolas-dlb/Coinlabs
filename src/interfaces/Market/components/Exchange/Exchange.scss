.exchange {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 20px;
  background: var(--third-bg-color);

  width: 93%;

  .exchange-top {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 3vh 0;
    width: 80vw;

    .exchange_left {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      margin-right: 20px;
      color: var(--third-font-color);
      width: 65%;

      p {
        align-self: flex-start;
        color: #c4c3c9;
      }
      .input {
        display: flex;
        justify-content: space-between;
        align-items: flex-start;
        border-bottom: 1px solid var(--third-font-color);
        padding-bottom: 1.3vh;
        width: 100%;
        margin-top: 1vh;
        height: 25px;
        &:first-of-type {
          margin-bottom: 20px;
        }

        input {
          background-color: var(--third-bg-color);
          color: var(--primary-font-color);
          outline: none;
          border: none;
          font-family: Karla;
          width: 100%;
          font-size: 18px;
          caret-color: #1baa63;
          &::-webkit-outer-spin-button,
          &::-webkit-inner-spin-button {
            -webkit-appearance: none;
            margin: 0;
          }

          &[type="number"] {
            -moz-appearance: textfield;
          }
        }
      }
    }
    .exchange_right {
      background: linear-gradient(
        137deg,
        rgba(10, 171, 81, 1) 0%,
        rgba(30, 169, 156, 1) 100%
      );
      pointer-events: none;
      border-radius: 20px;
      height: 140px;
      padding-left: 12px;
      padding-top: 1.5vh;
      width: 90px;
      svg {
        width: 30px;
      }
      p:first-of-type {
        font-size: 14px;
        margin-top: 1vh;
      }
      .wallet {
        font-size: 14px;
        font-weight: bold;
        margin: 1.5vh 0;
        position: relative;
        top: 3px;
        display: flex;
        p:first-child {
          position: relative;
          bottom: 7px;
          margin-right: 1px;
        }
      }
      .date {
        font-size: 14px;
      }
    }
  }
  .btn-exchange {
    padding: 12px 8px;
    border-radius: 10px;
    color: var(--primary-font-color);
    background: #373447;
    outline: none;
    border: none;
    width: 85%;
    font-size: 16px;
    font-family: Karla;
    margin-bottom: 15px;
    cursor: pointer;
  }
}
@media only screen and (max-height: 600px) {
  .input {
    margin-top: 0.7vh !important;
  }
  .exchange_right {
    height: 125px !important;
  }
  .btn-exchange {
    padding: 8px 15px !important;
  }
}

@media only screen and (min-width: 992px) {
  .exchange-top {
    width: 330px !important;
    .exchange_right {
      width: 100px !important;
      .wallet {
        font-size: 16px !important;
      }
    }
  }
}
