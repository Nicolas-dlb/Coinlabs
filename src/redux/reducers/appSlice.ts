import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  search: "Bitcoin",
};

const appSlice = createSlice({
  name: "app",
  initialState,
  reducers: {
    setSearch: (state, action) => {
      state.search = action.payload;
    },
  },
});
export const selectSearch = (state: any) => state.app.search;
export const { setSearch } = appSlice.actions;
export default appSlice.reducer;
