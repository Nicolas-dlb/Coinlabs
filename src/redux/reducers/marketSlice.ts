import { createSlice } from "@reduxjs/toolkit";

type CryptoState = {
  market: object;
  history: any;
  variation: object;
  time: string;
};
const initialState: CryptoState = {
  market: {},
  history: {},
  variation: { ethereum: 0, bitcoin: 0, ripple: 0, litecoin: 0, neo: 0 },
  time: "month",
};

const marketSlice = createSlice({
  name: "market",
  initialState,
  reducers: {
    loadCrypto: (state: CryptoState, action: any) => {
      state.market = { ...action.payload };
    },
    loadHistory: (state: CryptoState, action: any) => {
      state.history = { ...state.history, ...action.payload };
    },
    setVariation: (state: CryptoState, action: any) => {
      state.variation = { ...state.variation, ...action.payload };
    },
    setTime: (state: CryptoState, action: any) => {
      state.time = action.payload;
    },
  },
});

export const selectMarket = (state: any) => state.market.market;
export const selectHistory = (state: any) => state.market.history;
export const selectVariation = (state: any) => state.market.variation;
export const selectTime = (state: any) => state.market.time;
export const { loadCrypto, loadHistory, setVariation, setTime } =
  marketSlice.actions;
export default marketSlice.reducer;
