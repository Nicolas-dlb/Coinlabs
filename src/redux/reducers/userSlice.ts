import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  userName: null,
  userEmail: null,
  profilPic: null,
};

const userSlice = createSlice({
  name: "user",
  initialState,
  reducers: {
    login: (state, action) => {
      state.userName = action.payload.userName;
      state.userEmail = action.payload.email;
    },
    logout: (state) => {
      state.userName = null;
      state.userEmail = null;
    },
    setUsername: (state, action) => {
      state.userName = action.payload;
    },
    setProfilPic: (state, action) => {
      state.profilPic = action.payload;
    },
    resetProfilPic: (state) => {
      state.profilPic = null;
    },
  },
});
export const selectUserName = (state: any) => state.user.userName;
export const selectUserEmail = (state: any) => state.user.userEmail;
export const selectProfilPic = (state: any) => state.user.profilPic;
export const { login, logout, setUsername, setProfilPic, resetProfilPic } =
  userSlice.actions;
export default userSlice.reducer;
