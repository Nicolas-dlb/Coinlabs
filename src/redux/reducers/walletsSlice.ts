import { createSlice } from "@reduxjs/toolkit";

type WalletsState = {
  wallets: any;
  balanceHistory: any;
  walletHistory: any;
  cryptoHistory: any;
  incomeHistory: any;
  expensesHistory: any;
};
const initialState: WalletsState = {
  wallets: {
    usd: 20000,
    eur: 0,
    gbp: 0,
    mxn: 0,
    ethereum: 0,
    ethereumPrice: 0,
    neo: 0,
    neoPrice: 0,
    litecoin: 0,
    litecoinPrice: 0,
    bitcoin: 0,
    bitcoinPrice: 0,
    ripple: 0,
    ripplePrice: 0,
    TotalCrypto: 0,
    total: 20000,
  },
  balanceHistory: [],
  walletHistory: [],
  cryptoHistory: [],
  incomeHistory: [],
  expensesHistory: [],
};

function isIterable(value: any) {
  return Symbol.iterator in Object(value);
}

const walletsSlice = createSlice({
  name: "wallets",
  initialState,
  reducers: {
    loadWallets: (state: WalletsState, action: any) => {
      state.wallets = {
        ...state.wallets,
        ...action.payload,
        eur: state.wallets.usd * 0.82366545,
        gbp: state.wallets.usd * 0.710764,
        mxn: state.wallets.usd * 19.945724,
      };
    },
    setWallets: (state: WalletsState, action: any) => {
      state.wallets = action.payload.usd
        ? {
            ...state.wallets,
            ...action.payload,

            eur: action.payload.usd * 0.82366545,
            gbp: action.payload.usd * 0.710764,
            mxn: action.payload.usd * 19.945724,
          }
        : {
            ...state.wallets,
            ...action.payload,
          };
    },
    setBalanceHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.balanceHistory)) {
        state.balanceHistory = [...state.balanceHistory, action.payload];
      } else {
        state.balanceHistory = [action.payload];
      }
    },
    loadBalanceHistory: (state: WalletsState, action: any) => {
      state.balanceHistory = action.payload;
    },
    resetBalanceHistory: (state: WalletsState) => {
      state.balanceHistory = [];
    },
    setWalletHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.walletHistory)) {
        state.walletHistory = [action.payload, ...state.walletHistory];
      } else {
        state.walletHistory = [action.payload];
      }
    },
    resetWalletHistory: (state: WalletsState) => {
      state.walletHistory = [];
    },
    loadWalletHistory: (state: WalletsState, action: any) => {
      state.walletHistory = action.payload;
    },
    setCryptoHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.cryptoHistory)) {
        state.cryptoHistory = [...state.cryptoHistory, action.payload];
      } else {
        state.cryptoHistory = [action.payload];
      }
    },
    loadCryptoHistory: (state: WalletsState, action: any) => {
      state.cryptoHistory = action.payload;
    },
    resetCryptoHistory: (state: WalletsState) => {
      state.cryptoHistory = [];
    },
    setIncomeHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.incomeHistory)) {
        state.incomeHistory = [...state.incomeHistory, action.payload];
      } else {
        state.incomeHistory = [action.payload];
      }
    },
    loadIncomeHistory: (state: WalletsState, action: any) => {
      state.incomeHistory = action.payload;
    },
    resetIncomeHistory: (state: WalletsState) => {
      state.incomeHistory = [];
    },
    setExpensesHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.expensesHistory)) {
        state.expensesHistory = [...state.expensesHistory, action.payload];
      } else {
        state.expensesHistory = [action.payload];
      }
    },
    loadExpensesHistory: (state: WalletsState, action: any) => {
      state.expensesHistory = action.payload;
    },
    resetExpensesHistory: (state: WalletsState) => {
      state.expensesHistory = [];
    },
    
  },
});

export const selectWallets = (state: any) => state.wallets.wallets;
export const selectBalanceHistory = (state: any) =>
  state.wallets.balanceHistory;
  export const selectWalletHistory = (state: any) => state.wallets.walletHistory;
export const selectCryptoHistory = (state: any) => state.wallets.cryptoHistory;
export const selectIncomeHistory = (state: any) => state.wallets.incomeHistory;
export const selectExpensesHistory = (state: any) => state.wallets.expensesHistory;
export const {
  loadWallets,
  setWallets,
  setBalanceHistory,
  loadBalanceHistory,
  resetBalanceHistory,
  setWalletHistory,
  resetWalletHistory,
  loadWalletHistory,
  setIncomeHistory,
  resetIncomeHistory,
  loadIncomeHistory,
  setExpensesHistory,
  resetExpensesHistory,
  loadExpensesHistory,
  setCryptoHistory,
  resetCryptoHistory,
  loadCryptoHistory,
} = walletsSlice.actions;
export default walletsSlice.reducer;
