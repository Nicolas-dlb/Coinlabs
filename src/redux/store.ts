import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
