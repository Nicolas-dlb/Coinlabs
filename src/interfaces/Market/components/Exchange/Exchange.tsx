import React, { useState } from "react";
import { getNumberFixed, lastUpdate, numberWithSpaces } from "utils/utils";
import SelectExchange from "components/SelectExchange/SelectExchange";
import $ from "jquery";
import {
  selectLastUpdate,
  selectWallets,
  setBalanceHistory,
  setExpensesHistory,
  setIncomeHistory,
  setWalletHistory,
  setWallets,
} from "redux/reducers/walletsSlice";
import { useDispatch, useSelector } from "react-redux";
import { selectMarket } from "redux/reducers/marketSlice";
import "./Exchange.scss";

function Exchange() {
  const wallet = useSelector(selectWallets);
  const lastUpdateDate = useSelector(selectLastUpdate);
  const dispatch = useDispatch();
  const [cryptoSelected, setCryptoSelected] = useState("BTC");
  const [currencySelected, setCurrencySelected] = useState("bitcoin");
  const [nbr, setNbr] = useState(0);
  const [nbrCurrency, setNbrCurrency] = useState(0);
  const [lastInput, setLastInput] = useState("");
  const market = useSelector(selectMarket);
  const numberCrypto = market[cryptoSelected]?.usd * nbr;
  const numberCurrency = nbrCurrency * market[currencySelected]?.usd;

  const handleChangeCurrency = () => {
    const inputBuy: any = $("#inputBuy").val();
    setNbrCurrency(Number(inputBuy));
  };
  const handleChangeCrypto = () => {
    const inputSell: any = $("#inputSell").val();
    setNbr(Number(inputSell));
  };

  const handleClick = () => {
    if (Number(nbr) === 0 && Number(nbrCurrency) === 0) {
      document
        .getElementById("inputBuy")!
        .classList.add("bad_input_send_crypto");
      setTimeout(() => {
        document
          .getElementById("inputBuy")!
          .classList.remove("bad_input_send_crypto");
      }, 5000);
      document
        .getElementById("inputSell")!
        .classList.add("bad_input_send_crypto");
      setTimeout(() => {
        document
          .getElementById("inputSell")!
          .classList.remove("bad_input_send_crypto");
      }, 5000);
    } else if (lastInput === "crypto") {
      if (Number(nbr) > wallet[cryptoSelected] || Number(nbr) === 0) {
        document
          .getElementById("inputSell")!
          .classList.add("bad_input_send_crypto");
        setTimeout(() => {
          document
            .getElementById("inputSell")!
            .classList.remove("bad_input_send_crypto");
        }, 5000);
        return;
      }

      dispatch(
        setWallets({
          [cryptoSelected]: wallet[cryptoSelected] - Number(nbr),

          usd: wallet.usd + Number(getNumberFixed(numberCrypto, 2)),
          income: Number(getNumberFixed(wallet.income + numberCrypto, 2)),
        })
      );

      dispatch(
        setBalanceHistory({
          total: wallet.usd + Number(numberCrypto),
          timestamp: new Date().getTime(),
        })
      );

      dispatch(
        setIncomeHistory({
          total: Number(getNumberFixed(wallet.income + numberCrypto, 2)),
          timestamp: new Date().getTime(),
        })
      );

      dispatch(
        setWalletHistory({
          status: "Sell",
          name:
            cryptoSelected.charAt(0).toUpperCase() + cryptoSelected.slice(1),

          price: getNumberFixed(Number(numberCrypto), 0),
        })
      );
      lastUpdate(dispatch);
      setLastInput("");
      setNbr(0);
      $("#inputSell").val(0);
    } else if (lastInput === "currency") {
      if (Number(numberCurrency) > wallet.usd || Number(numberCurrency) === 0) {
        document
          .getElementById("inputBuy")!
          .classList.add("bad_input_send_crypto");
        setTimeout(() => {
          document
            .getElementById("inputBuy")!
            .classList.remove("bad_input_send_crypto");
        }, 5000);
        return;
      }

      dispatch(
        setBalanceHistory({
          total: wallet.usd - Number(numberCurrency),
          timestamp: new Date().getTime(),
        })
      );
      dispatch(
        setWallets({
          [currencySelected]: wallet[currencySelected] + Number(nbrCurrency),

          usd: Number(getNumberFixed(wallet.usd - Number(numberCurrency), 2)),
          expenses: Number(getNumberFixed(wallet.expenses + numberCurrency, 2)),
        })
      );

      const timestamp = new Date().getTime();
      dispatch(
        setExpensesHistory({
          total: Number(getNumberFixed(wallet.expenses + numberCurrency, 2)),
          timestamp,
        })
      );
      dispatch(
        setWalletHistory({
          status: "Buy",
          name:
            currencySelected.charAt(0).toUpperCase() +
            currencySelected.slice(1),

          price: getNumberFixed(Number(numberCurrency), 0),
        })
      );
      lastUpdate(dispatch);

      setLastInput("");
      setNbrCurrency(0);
      $("#inputBuy").val(0);
    }
  };

  document.addEventListener("click", (e: any) => {
    if (!e.target.classList.contains("exchange_element")) {
      setNbr(1);
      setNbrCurrency(1);
      $("#inputSell").val(0);
      $("#inputBuy").val(0);
      $("#inputSell").css("opacity", "0");
      $("#inputBuy").css("opacity", "0");
      setLastInput("");
    }
  });
  return (
    <div className="exchange exchange_element">
      <div className="exchange-top exchange_element">
        <div className="exchange_left exchange_element">
          <p className="exchange_element">Sell</p>
          <div className="input exchange_element">
            <input
              className="exchange_element"
              onFocus={() => {
                $("#inputBuy").val(0);
                $("#inputBuy").css("opacity", "0");
                $("#inputSell").css("opacity", "1");
                setNbrCurrency(0);
                setLastInput("crypto");
              }}
              onChange={() => {
                document
                  .getElementById("inputBuy")!
                  .classList.remove("bad_input_send_crypto");
                document
                  .getElementById("inputSell")!
                  .classList.remove("bad_input_send_crypto");
                handleChangeCrypto();
              }}
              min="0"
              type="number"
              id="inputSell"
            />
            <SelectExchange
              setCryptoSelected={setCryptoSelected}
              items="sell"
            />
          </div>
          <p className="exchange_element">Buy</p>
          <div className="input exchange_element">
            <input
              className="exchange_element"
              onFocus={() => {
                $("#inputSell").val(0);
                $("#inputSell").css("opacity", "0");
                $("#inputBuy").css("opacity", "1");
                setNbr(0);
                setLastInput("currency");
              }}
              min="0"
              onChange={() => {
                document
                  .getElementById("inputBuy")!
                  .classList.remove("bad_input_send_crypto");
                document
                  .getElementById("inputSell")!
                  .classList.remove("bad_input_send_crypto");
                handleChangeCurrency();
              }}
              type="number"
              id="inputBuy"
            />
            <SelectExchange
              setCurrencySelected={setCurrencySelected}
              items="buy"
            />
          </div>
        </div>
        <div className="exchange_right">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 66 65"
            fill="#fff"
            fillRule="evenodd"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
          >
            <path
              d="M31.848 0c4.993 0 9.744 1.144 14.007 3.2.67.302 1.34.663 2.008 1.024L44.94 6.813l-4.382-4.515-7.55 6.62-4.448-4.765-13.885 12.342 8.9 9.692-3.47 3 8.77 9.692-3.47 3.008L37.94 55.683l7.425-6.743 6.458 7.226c-1.28.963-2.68 1.928-4.2 2.77a32.11 32.11 0 0 1-15.712 4.09C14.3 63.026 0 48.883 0 31.486-.062 14.207 14.25 0 31.848 0zM26 29.558l7.612-6.743 6.822 7.525-7.614 6.742zm5.42 13.003l7.55-6.802 6.882 7.524-7.613 6.743zM20.584 16.736l7.6-6.743 6.82 7.525-7.613 6.742zm13.943-3.623L40.2 8.067l5.116 5.6-5.664 5.116zm10.1 23.177l5.673-5.047 5.116 5.657-5.664 5.06zm4.993 11.38l5.664-5.058 5.112 5.6-5.662 5.116zm2.315-26.007l3.775-3.372 3.4 3.734-3.774 3.37zm-4.872-11.137l3.775-3.43 3.412 3.792-3.778 3.37zm9.682 22.275l3.775-3.37 3.48 3.74-3.835 3.37zm-17-8.368l5.664-5.057 5.114 5.598-5.662 5.058z"
              stroke="none"
              fill="#fff"
              fillRule="nonzero"
            />
          </svg>
          <p>Balance</p>
          <div className="wallet">
            <p>$</p>
            <p>{numberWithSpaces(getNumberFixed(wallet.usd, 2))}</p>
          </div>
          <p className="date">{lastUpdateDate}</p>
        </div>
      </div>
      <button
        onClick={handleClick}
        className="btn-exchange exchange_element"
        type="button"
      >
        Exchange
      </button>
    </div>
  );
}

export default Exchange;
