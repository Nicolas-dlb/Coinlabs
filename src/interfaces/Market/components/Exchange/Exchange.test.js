import React from "react";
import { mount } from "enzyme";
import { act } from "react-test-renderer";
import store from "redux/store";
import { Provider } from "react-redux";
import { setWallets } from "redux/reducers/walletsSlice";
import Exchange from "./Exchange";

describe("Exchange", () => {
  let ExchangeComponent;
  beforeEach(async () => {
    await act(async () => {
      ExchangeComponent = mount(
        <Provider store={store}>
          <Exchange />
        </Provider>
      );
    });
  });
  it("should render correctly", () => {
    expect(ExchangeComponent).toMatchSnapshot();
  });

  it("should render 2 input", () => {
    expect(ExchangeComponent.find("input").length).toBe(2);
  });

  it("should render 2 Select components", () => {
    expect(ExchangeComponent.find(".sell").first().exists()).toBe(true);
    expect(ExchangeComponent.find(".buy").first().exists()).toBe(true);
  });

  it("should render exchange right card", () => {
    expect(ExchangeComponent.find(".exchange_right").first().exists()).toBe(
      true
    );
  });

  it("should render correct current balance", () => {
    store.dispatch(setWallets({ usd: 100 }));
    expect(ExchangeComponent.find(".wallet").find("p").last().text()).toBe(
      "100.00"
    );
  });

  it("should render 1 button", () => {
    expect(ExchangeComponent.find(".btn-exchange").length).toBe(1);
  });
});
