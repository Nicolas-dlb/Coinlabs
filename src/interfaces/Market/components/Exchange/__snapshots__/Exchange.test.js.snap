// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Exchange>
    <div
      className="exchange exchange_element"
    >
      <div
        className="exchange-top exchange_element"
      >
        <div
          className="exchange_left exchange_element"
        >
          <p
            className="exchange_element"
          >
            Sell
          </p>
          <div
            className="input exchange_element"
          >
            <input
              className="exchange_element"
              id="inputSell"
              min="0"
              onChange={[Function]}
              onFocus={[Function]}
              type="number"
            />
            <Select
              items="sell"
              setCryptoSelected={[Function]}
            >
              <div>
                <select
                  className="sell exchange_element"
                >
                  <option
                    className="option exchange_element"
                    label="BTC"
                    value="BTC"
                  >
                    BTC
                  </option>
                  <option
                    className="option exchange_element"
                    label="ETH"
                    value="ETH"
                  >
                    ETH
                  </option>
                  <option
                    className="option exchange_element"
                    label="XRP"
                    value="XRP"
                  >
                    XRP
                  </option>
                  <option
                    className="option exchange_element"
                    label="LTC"
                    value="LTC"
                  >
                    LTC
                  </option>
                  <option
                    className="option exchange_element"
                    label="NEO"
                    value="NEO"
                  >
                    NEO
                  </option>
                </select>
                <div
                  className="sell-select exchange_element"
                >
                  <svg
                    className="select-sell-arrow exchange_element"
                    fill="#605b95"
                    height="24px"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                  <button
                    aria-label="select-sell"
                    className="btn-sell exchange_element"
                    id="select-sell"
                    type="button"
                    value=""
                  />
                  <div
                    className="sell-list exchange_element"
                  >
                    <ul
                      id="sell-items"
                    />
                  </div>
                </div>
              </div>
            </Select>
          </div>
          <p
            className="exchange_element"
          >
            Buy
          </p>
          <div
            className="input exchange_element"
          >
            <input
              className="exchange_element"
              id="inputBuy"
              min="0"
              onChange={[Function]}
              onFocus={[Function]}
              type="number"
            />
            <Select
              items="buy"
              setCurrencySelected={[Function]}
            >
              <div>
                <select
                  className="buy exchange_element"
                >
                  <option
                    className="option exchange_element"
                    label="BTC"
                    value="BTC"
                  >
                    BTC
                  </option>
                  <option
                    className="option exchange_element"
                    label="ETH"
                    value="ETH"
                  >
                    ETH
                  </option>
                  <option
                    className="option exchange_element"
                    label="XRP"
                    value="XRP"
                  >
                    XRP
                  </option>
                  <option
                    className="option exchange_element"
                    label="LTC"
                    value="LTC"
                  >
                    LTC
                  </option>
                  <option
                    className="option exchange_element"
                    label="NEO"
                    value="NEO"
                  >
                    NEO
                  </option>
                </select>
                <div
                  className="buy-select exchange_element"
                >
                  <svg
                    className="select-buy-arrow exchange_element"
                    fill="#605b95"
                    height="24px"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                    <path
                      d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                  <button
                    aria-label="select-buy"
                    className="btn-buy exchange_element"
                    id="select-buy"
                    type="button"
                    value=""
                  />
                  <div
                    className="buy-list exchange_element"
                  >
                    <ul
                      id="buy-items"
                    />
                  </div>
                </div>
              </div>
            </Select>
          </div>
        </div>
        <div
          className="exchange_right"
        >
          <svg
            fill="#fff"
            fillRule="evenodd"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            viewBox="0 0 66 65"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.848 0c4.993 0 9.744 1.144 14.007 3.2.67.302 1.34.663 2.008 1.024L44.94 6.813l-4.382-4.515-7.55 6.62-4.448-4.765-13.885 12.342 8.9 9.692-3.47 3 8.77 9.692-3.47 3.008L37.94 55.683l7.425-6.743 6.458 7.226c-1.28.963-2.68 1.928-4.2 2.77a32.11 32.11 0 0 1-15.712 4.09C14.3 63.026 0 48.883 0 31.486-.062 14.207 14.25 0 31.848 0zM26 29.558l7.612-6.743 6.822 7.525-7.614 6.742zm5.42 13.003l7.55-6.802 6.882 7.524-7.613 6.743zM20.584 16.736l7.6-6.743 6.82 7.525-7.613 6.742zm13.943-3.623L40.2 8.067l5.116 5.6-5.664 5.116zm10.1 23.177l5.673-5.047 5.116 5.657-5.664 5.06zm4.993 11.38l5.664-5.058 5.112 5.6-5.662 5.116zm2.315-26.007l3.775-3.372 3.4 3.734-3.774 3.37zm-4.872-11.137l3.775-3.43 3.412 3.792-3.778 3.37zm9.682 22.275l3.775-3.37 3.48 3.74-3.835 3.37zm-17-8.368l5.664-5.057 5.114 5.598-5.662 5.058z"
              fill="#fff"
              fillRule="nonzero"
              stroke="none"
            />
          </svg>
          <p>
            Balance
          </p>
          <div
            className="wallet"
          >
            <p>
              $
            </p>
            <p>
              20 000.00
            </p>
          </div>
          <p
            className="date"
          >
            00/00
          </p>
        </div>
      </div>
      <button
        className="btn-exchange exchange_element"
        onClick={[Function]}
        type="button"
      >
        Exchange
      </button>
    </div>
  </Exchange>
</Provider>
`;
